plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2115,6 +2120,28 @@ private static boolean fieldExists(StructType structType, String fieldName)
         return false;
     }
 
+    private static void checkDuplicatedFieldNames(io.trino.spi.type.Type type)
+    {
+        if (type instanceof ArrayType arrayType) {
+            checkDuplicatedFieldNames(arrayType.getElementType());
+        }
+        else if (type instanceof MapType mapType) {
+            checkDuplicatedFieldNames(mapType.getKeyType());
+            checkDuplicatedFieldNames(mapType.getValueType());
+        }
+        if (type instanceof RowType rowType) {
```

Comment:
be consistent whether `else if` or separate `if`s
imo here `else if` would be more appropriate